crates/turborepo-lib/src/task_graph/visitor.rs

```diff
@@ -190,6 +193,13 @@ impl<'a> Visitor<'a> {
                 continue;
             }
 
+            // We do this calculation earlier than we do in Go due to the `task_hasher`
+            // being !Send In the future we can look at doing this right before
```

Comment:
`Send` (and `Sync`) are special traits in Rust as they're automatically implemented on types if all members are `Send` (i.e. if `T: Send` then `struct Foo { inner: T}` will also implement `Send`. 

`!Send` in this case just means that `Send` isn't implemented.

I find this chunk from the Rust docs useful for thinking about `Send` vs `!Send`:
> An example of a non-Send type is the reference-counting pointer [rc::Rc](https://doc.rust-lang.org/std/rc/struct.Rc.html). If two threads attempt to clone [Rc](https://doc.rust-lang.org/std/rc/struct.Rc.html)s that point to the same reference-counted value, they might try to update the reference count at the same time, which is [undefined behavior](https://doc.rust-lang.org/reference/behavior-considered-undefined.html) because [Rc](https://doc.rust-lang.org/std/rc/struct.Rc.html) doesn’t use atomic operations. Its cousin [sync::Arc](https://doc.rust-lang.org/std/sync/struct.Arc.html) does use atomic operations (incurring some overhead) and thus is Send.

[Rust docs](https://doc.rust-lang.org/std/marker/trait.Send.html). If you really want to get your hands dirty the Rustonomicon there's a [pretty detailed page on `Send + Sync`](https://doc.rust-lang.org/nomicon/send-and-sync.html)